internal/test/wmcb/pkginfo.go

```diff
@@ -0,0 +1,110 @@
+package wmcb
+
+import (
+	"bufio"
+	"fmt"
+	"net/http"
+	"strings"
+)
+
+// pkgName is the user defined name of the package
+type pkgName string
+
+//cniPlugins contains information about the CNI plugin package
+type cniPlugins struct {
+	pkgInfo
+}
+
+// PkgInfo is an interface to populate pkgInfo structs
+type PkgInfo interface {
+	//initUrl retrieves package url for desired version
+	initUrl(baseUrl string, version string) error
+	//initSHAValue retrieves SHA value from url and sets it to the sha field
+	initSHA() error
+}
+
+// pkgInfo encapsulates information about a package
+type pkgInfo struct {
+	// name of the package
+	name pkgName
+	// url is the URL of the package
+	url string
+	// sha is the SHA hash of the package
+	sha string
+	// shaType is the type of SHA used, example: 256 or 512
+	shaType string
+}
+
+// pkgInfoFactory returns PkgInfo specific to the package name
+func pkgInfoFactory(name pkgName) (PkgInfo, error) {
+	switch name {
+	case CniPluginPkgName:
+		// get url according to the latest cni plugins version
+		return &cniPlugins{pkgInfo{name: name, shaType: "sha512"}}, nil
+
+	default:
+		return nil, fmt.Errorf("invalid Package name")
+	}
+}
+
+// initUrl retrieves url of the given version of cni plugin and sets it to url field of cniPlugins
+func (c *cniPlugins) initUrl(baseUrl string, version string) error {
+	if version == "" {
+		return fmt.Errorf("unable to retrieve latest version of cni plugins")
+	}
+	c.url = baseUrl + version +
+		"/cni-plugins-windows-amd64-" + version + ".tgz"
+	return nil
+}
+
+// initSHA retrieves SHA value for the given url and sets it to sha field of cniPlugins
+func (c *cniPlugins) initSHA() error {
+	checksumFileContent, err := getSHAFileContent(c.url)
+	if err != nil {
+		return err
+	}
+	// The checksum file content is in the format "<sha> <filename>". So to get SHA we need to extract only the <sha>
+	// from the file
+	sha512 := strings.Split(checksumFileContent, " ")
+	if len(sha512) < 2 {
+		return fmt.Errorf("checksum file content is not in the format : '<sha>'")
+	}
+	c.sha = sha512[0]
+	return nil
+}
+
+// initializePkg sets all the fields for given pkg name and returns the
+func initializePkg(name pkgName, url string, version string) (PkgInfo, error) {
+	// get pkg instance for given package name
+	pkg, err := pkgInfoFactory(name)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create pkgInfo instance for given package : %v", err)
+	}
+	// retrieve latest url and set it to the url field
+	if err := pkg.initUrl(url, version); err != nil {
+		return nil, fmt.Errorf("error retrieving the url value for the package: %v", err)
+	}
+	// retrieve SHA value and set it to the sha field
+	if err := pkg.initSHA(); err != nil {
+		return nil, fmt.Errorf("error retrieving the SHA value for the package: %v", err)
+	}
+	return pkg, nil
+}
+
+// getSHAFileContent returns the contents of the SHA file for the given url of the package.
+func getSHAFileContent(url string) (string, error) {
```

Comment:
This should be associated with `pkgInfo` and when you do that, please check that `url != ""`.